### ISSUE TITLE
Unexpected SC2034 warning

### ISSUE THREAD

**Issue by pcolmer** [koalaman/shellcheck#1660]
#### For bugs
- Rule Id (if any, e.g. SC1000): SC2034
- My shellcheck version (`shellcheck --version` or "online"): 0.4.6 and online
- [x] The rule's wiki page does not already cover this (e.g. https://shellcheck.net/wiki/SC2086)
- [x] I tried on shellcheck.net and verified that this is still a problem on the latest commit

#### Here's a snippet or screenshot that shows the problem:

```sh

#!/bin/bash
process_repos() {
    # If we already have a "merged_sources" directory, clean it out.
    if [ -d "/srv/source/merged_sources" ]; then
        rm -r /srv/source/merged_sources
    fi
    # Iterate through the manifest and process each repo in turn.
    while IFS=, read -r varname, giturl, dirpath
    do
        process_single_repo varname giturl dirpath
    done < /srv/source/manifest.csv
}

```

#### Here's what shellcheck currently says:

dirpath appears unused. Verify it or export it.

#### Here's what I wanted or expected to see:

Nothing. All three variables are assigned values from the read command and then passed as values to process_single_repo, so all three should be seen as being used.

**Comment by Nightfirecat** [comment]
Aren't you just passing string literals to `process_single_repo`? That is, I think your code is the same as the following:

```sh
#!/bin/bash
process_repos() {
    # If we already have a "merged_sources" directory, clean it out.
    if [ -d "/srv/source/merged_sources" ]; then
        rm -r /srv/source/merged_sources
    fi
    # Iterate through the manifest and process each repo in turn.
    while IFS=, read -r varname, giturl, dirpath
    do
        process_single_repo 'varname' 'giturl' 'dirpath'
    done < /srv/source/manifest.csv
}
```

**Comment by pcolmer** [comment]
Yeah, there are a number of mistakes in my code :).

It actually needed to read:

```
    while IFS=, read -r varname giturl dirpath
    do
        process_single_repo "$varname" "$giturl" "$dirpath"
    done < /srv/source/manifest.csv
```
and then shellcheck is happy, and so am I.

Sorry about the noise.
